OMCompiler/Compiler/FrontEnd/AbsynUtil.mo

```diff
@@ -1530,6 +1530,7 @@ public function pathRest
   output Absyn.Path outPath;
 algorithm
   outPath := match inPath
+    case Absyn.IDENT(_) then inPath;
```

Comment:
This is wrong, `pathRest` of a single identifier path is the empty path, but `Absyn.Path` can't be empty. Returning the same path in that case will give the wrong result.

Reply:
Without this it fails for the following example,

```
loadFile("model InstantiationExample
  block MyBlock
    extends Modelica.Blocks.Icons.Block;
  end MyBlock;
end InstantiationExample;");
getErrorString();
list(InstantiationExample);
addComponent(myBlock, InstantiationExample.MyBlock,InstantiationExample,annotate=Placement(transformation=transformation(origin={-32,-62},extent={{-10,-10},{10,10}})));
getErrorString();
list(InstantiationExample);
getErrorString();
```

Reply:
Yes, I guess `pathStripSamePrefix` needs to be improved to handle the case where the path being compared against is shorter. I can see if I can rewrite it if you want. But the change to `pathRest` is definitely wrong regardless.

Reply:
I rewrote `pathStripSamePrefix` to handle this case, but since it's such a small change I just made my own PR for it (#12212) instead.